tests/virt_autotest/virt_utils.pm

```diff
@@ -330,12 +330,13 @@ sub get_guest_disk_name_from_guest_xml {
 # Please refer to https://qemu.readthedocs.io/en/latest/tools/qemu-img.html to ease your mind on working with --force-share and convert and many others.
 sub compress_single_qcow2_disk {
     my ($orig_disk, $compressed_disk) = @_;
+    my $timeout = is_kvm_host ? '360' : '480';
 
     if ($orig_disk =~ /qcow2/) {
         my $cmd = "nice ionice qemu-img convert -c -p -O qcow2 $orig_disk $compressed_disk";
-        if (script_run($cmd, 360) ne 0) {
+        if (script_run($cmd, $timeout) ne 0) {
             $cmd = "nice ionice qemu-img convert --force-share -c -p -O qcow2 $orig_disk $compressed_disk";
-            die("Disk compression failed from $orig_disk to $compressed_disk.") if (script_run($cmd, 360) ne 0);
+            die("Disk compression failed from $orig_disk to $compressed_disk.") if (script_run($cmd, $timeout) ne 0);
```

Comment:
Code looks good. Thanks for the fix.
Would you please give some comparison data, compression time during 15sp5 gmc vs 15sp6? Any performance related bug maybe worthwhile?

Reply:
Good point, I am working on this as you mentioned now.

Reply:
Quick update, after collected relevant comparison data from 12SP5, 15SP5 and 15SP6 vm host, found that it was a new linux kernel version 6.4 performance problem that timeout to compress guest system disk image via qemu-img in 15SP6 build28.1 vm host.

Please refer to the following information for more details:
Comparison on different versions of SLE using the same machine - gonzo-1 
(CPU:Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz, Memory: 62G)

Figure out that start to use with linux kernel 6.4.0 from 15SP6 build28.1 - SLE-15-SP6-Full-Alpha-202310-2.
However, still run with linux kernel 5.14.21 until 15SP6 build26.1 - SLE-15-SP6-Full-Alpha-202310-2
Meanwhile, there is the Zstandard as the default File System Compression Algorithm in kernel 6.4.0, but, kernel 5.14 still use LZ4 as the default one.

So, supposed that it was Kernel 6.4 performance problem for Zstandard as the default File System Compression Algorithm.  File a new product issue - [bsc#1216796](https://bugzilla.suse.com/show_bug.cgi?id=1216796) - [15SP6][KERNEL][6.4] Timeout to compress guest system disk image via qemu-img

15SP6 build28.1KVM|XEN system role
gonzo-1:/var/lib/libvirt/images # time qemu-img convert --force-share -c -p -O qcow2 sles-15-sp6-64-fv-def-net.qcow2 sles-15-sp6-64-fv-def-net.disk
    (100.00/100%)

real    9m30.263s
user    9m13.282s
sys 0m15.166s

15SP5 Update maintain KVM|XEN system role
gonzo-1:~ # time qemu-img convert --force-share -c -p -O qcow2 sles-15-sp6-64-fv-def-net.qcow2 sles-15-sp6-64-fv-def-net.disk
    (100.00/100%)

real    3m35.902s
user    3m23.259s
sys 0m7.318s    

12SP5 Update maintain KVM|XEN system role
gonzo-1:~ # time qemu-img convert --force-share -c -p -O qcow2 sles-15-sp6-64-fv-def-net.qcow2 sles-15-sp6-64-fv-def-net.disk
    (100.00/100%)

real    3m43.820s
user    3m33.807s
sys 0m7.724s